tests/org.jboss.tools.aerogear.ui.bot.test/src/org/jboss/tools/aerogear/ui/bot/test/AerogearBotTest.java

```diff
@@ -54,244 +52,250 @@
+
+		// TODO: Order/content of context many may change
+		// TODO: Need to check presence of Android SDK installation
+		getProjectExplorer().selectProjects(projectName);
+		new ContextMenu(new WithTextMatcher("Run As"), new RegexMatcher("(\\d+)( Run on Android Emulator)")).select();
+		new WaitWhile(new JobIsRunning());
 	}
-	
-	//workaround for https://bugs.eclipse.org/bugs/show_bug.cgi?id=506123
-	//save config.xml
-	DefaultEditor configEditor = new DefaultEditor(CORDOVA_APP_NAME);
-	configEditor.activate();
-	configEditor.save();
-	
-  }
 
-  public void runTreeItemInAndroidEmulator(String projectName) {
-    
-    // TODO: Order/content of context many may change
-    // TODO: Need to check presence of Android SDK installation
-    new DefaultTreeItem(projectName).select();
-    new ContextMenu(new WithTextMatcher("Run As"), new RegexMatcher("(\\d+)( Run on Android Emulator)")).select();
-    new WaitWhile(new JobIsRunning());
-  }
+	public void runTreeItemOnAndroidDevice(String projectName) {
+
+		// TODO: Order/content of context many may change
+		// TODO: Need to check presence of Android SDK installation
+		getProjectExplorer().selectProjects(projectName);
+		new ContextMenu(new WithTextMatcher("Run As"), new RegexMatcher("(\\d+)( Run on Android Device)")).select();
+		new WaitWhile(new JobIsRunning());
+	}
+
+	public void runTreeItemWithCordovaSim(String projectName) {
+		getProjectExplorer().selectProjects(projectName);
+		new ContextMenu(new WithTextMatcher("Run As"), new RegexMatcher("(\\d+)( Run w/CordovaSim)")).select();
+		new WaitWhile(new JobIsRunning());
+		new WaitUntil(new ConsoleHasNoChange(TimePeriod.LONG), TimePeriod.VERY_LONG);
+	}
 
-  public void runTreeItemOnAndroidDevice(String projectName) {
-    
-    // TODO: Order/content of context many may change
-    // TODO: Need to check presence of Android SDK installation
-    new DefaultTreeItem(projectName).select();
-    new ContextMenu(new WithTextMatcher("Run As"), new RegexMatcher("(\\d+)( Run on Android Device)")).select();
-    new WaitWhile(new JobIsRunning());
-  }
+	/**
+	 * Opens config.xml in Cordova Configuration Editor
+	 * 
+	 * @param workspace
+	 *            related path to config.xml file
+	 */
+	public void openInConfigEditor(String projectName, String appName, String... path) {
+		getProjectExplorer().getProject(projectName).getProjectItem(path).open();
+		// check if correct editor was opened
+		new ConfigEditor(appName);
+		new WaitWhile(new JobIsRunning());
+	}
 
-  public void runTreeItemWithCordovaSim(String projectName) {
+	/**
+	 * @Before public void setUp() {
+	 *         createHTMLHybridMobileApplication(AerogearBotTest.CORDOVA_PROJECT_NAME,
+	 *         AerogearBotTest.CORDOVA_APP_NAME, "org.jboss.example.cordova",
+	 *         "cordova-android@4.1.0");
+	 * 
+	 *         assertTrue(new
+	 *         ProjectExplorer().containsProject(AerogearBotTest.CORDOVA_PROJECT_NAME));
+	 *         }
+	 */
```

Comment:
We must have missed this commented code in PR review before. But it would be nice if you could remove it in your PR.